main.tf

```diff
@@ -34,6 +34,7 @@ resource "random_shuffle" "available_zones" {
 locals {
   // location
   location = var.regional ? var.region : var.zones[0]
+  region   = var.region != null ? var.region : join("-", slice(split("-", var.zones[0]), 0, 2))
```

Comment:
See above -- this should be taken care of by making the change above and re-generating.

Reply:
updated